### PR TITLE
Docker Local Source

### PR DESCRIPTION
**Pull request by intolerance**
This pull request adds support for building a Docker container using the local source code.

I had to move this out to the root directory because `docker build` command will not let you traverse outside of it's pwd (e.g. ../).

In addition to the Dockerfile, I also added a lot of detail to the `README` in the `/docker` directory.

I hope you will consider merging this PR -- if I need to update or clean up anything, please let me know.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,70 @@
+
+#Use debian:stable-slim as a builder and then copy everything.
+FROM debian:stable-slim as builder
+
+#Set mosquitto and plugin versions.
+#Change them for your needs.
+ENV MOSQUITTO_VERSION=1.6.3
```
Comment (iegomez) [inline Dockerfile]: Could you bump Mosquitto and Go versions to latest?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+RUN cd mosquitto-${MOSQUITTO_VERSION} && make WITH_WEBSOCKETS=yes && make install && cd ..
+
+#Get Go.
+RUN wget https://dl.google.com/go/go${GO_VERSION}.linux-amd64.tar.gz && tar -C /usr/local -xzf go${GO_VERSION}.linux-amd64.tar.gz
+RUN export PATH=$PATH:/usr/local/go/bin && go version && rm go${GO_VERSION}.linux-amd64.tar.gz
+
+
+# #Get the plugin.
+# RUN wget https://github.com/iegomez/mosquitto-go-auth/archive/${PLUGIN_VERSION}.tar.gz \
+#     && ls -l \
```
Comment (iegomez) [inline Dockerfile]: Could you delete the commented bits here and later on this file?

File: Dockerfile (hunk 1 above)
Comment (intolerance) [reply Dockerfile]: Done.

File: Dockerfile (hunk 2 above)
Comment (intolerance) [reply Dockerfile]: I removed this section. Was this all you were referring to?

**Pull request merged**